pkg/genericapiserver/options/server_run_options.go

```diff
@@ -55,6 +54,8 @@ var AuthorizationModeChoices = []string{ModeAlwaysAllow, ModeAlwaysDeny, ModeABA
 
 // ServerRunOptions contains the options while running a generic api server.
 type ServerRunOptions struct {
+	EtcdOptions *EtcdOptions
```

Comment:
> What's the reason to have a pointer here to EtcdOptions?

Instead of completely distinct structs?  Momentum for now.  I've got practice making two API servers, now, but I'm not completely sure what we should cut yet.

I *think* what I'm going to recommend is that every API server create its own options with whatever bits it needs, but I think a lot is going to hinge on the authentication/authorization bits.